plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/conversion/StmtConversionContext.kt

```diff
@@ -71,19 +76,60 @@ fun StmtConversionContext.declareLocalVariable(symbol: FirVariableSymbol<*>, ini
 fun StmtConversionContext.declareAnonVar(type: TypeEmbedding, initializer: ExpEmbedding?): Declare =
     Declare(freshAnonVar(type), initializer)
 
+
+val FirIntersectionOverridePropertySymbol.propertyIntersections
+    get() = intersections.filterIsInstanceAnd<FirPropertySymbol> { it.isVal == isVal }
+
+/**
+ * Tries to find final property symbol actually declared in some class instead of
+ * (potentially) fake property symbol.
+ * Note that if some property is found it is fixed since
+ * 1. there can't be two non-abstract properties which don't subsume each other
+ * in the hierarchy (kotlin disallows that) and final properties can't be abstract;
+ * 2. final property can't subsume other final property as that means final property
+ * is overridden.
+ */
+fun FirPropertySymbol.findFinalParentProperty(): FirPropertySymbol? =
+    if (this !is FirIntersectionOverridePropertySymbol)
+        isFinal.ifTrue { this }
+    else propertyIntersections.firstNotNullOfOrNull { it.findFinalParentProperty() }
+
+fun FirPropertySymbol.findSomeParentProperty(): FirPropertySymbol =
+    if (this is FirIntersectionOverridePropertySymbol)
+        propertyIntersections.first().findSomeParentProperty()
+    else this
```

Comment:
I'd like to understand what's going on here a bit better; why do we need to find the final property rather than working with the one that's here?

Reply:
Overall this is the part I'd like to discuss with you and Ilya during this week.

Reply:
Thanks, the comments help.  Let's discuss with Ilya.  My problem here is: we are doing some work to go from a specific piece of information we have, to some arbitrary choice.

One thing to note here: when a function takes an interface parameter, that can't possible contain any access permissions.  So for an interface property, we don't actually care which interface we take it from.  But perhaps we can do something better than taking it from a random one?  One idea is that we could have a common method for the actual call, for example, and then use the property symbol we have here to find the right access to inhale?